src/common/servers/MattermostServer.ts

```diff
@@ -1,14 +1,30 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {v4 as uuid} from 'uuid';
+
+import {Team} from 'types/config';
+
 import urlUtils from 'common/utils/url';
 
 export class MattermostServer {
+    id: string;
     name: string;
-    url: URL;
-    constructor(name: string, serverUrl: string) {
-        this.name = name;
-        this.url = urlUtils.parseURL(serverUrl)!;
+    url!: URL;
+    isPredefined: boolean;
+
+    constructor(server: Team, isPredefined = false) {
+        this.id = uuid();
```

Comment:
Well for one, I didn't know we had that :P.

In this case, the unique identifier is purely there to have a consistent and non-colliding ID so that we can ID a server from anywhere in the app, the actual string itself doesn't have any meaning otherwise.

I can see some usefulness in using the diagnostic ID, but when we create the server we don't have access to that yet, we'd have to ping the server first and see if we can get that. I think for the purposes I'm using the ID for right now the UUID is fine, but I could see a future change in which we store the diagnostic ID after the server has been added and switch to using that when we do logging.

Reply:
sounds good to me